source/aggregation.txt

```diff
@@ -0,0 +1,195 @@
+- Summarize data
+- Group values
+
+Limitations
+~~~~~~~~~~~
+
+Keep the following limitations in mind when using aggregation operations:
+
+- Returned documents must not violate the
+  :manual:`BSON document size limit </reference/limits/#mongodb-limit-BSON-Document-Size>`
+  of 16 megabytes.
+- Pipeline stages have a memory limit of 100 megabytes by default. You can exceed this
+  limit by setting the ``allow_disk_use`` field of a ``mongocxx::options::aggregate``
+  instance.
+
+.. important:: $graphLookup Exception
+
+   The :manual:`$graphLookup
+   </reference/operator/aggregation/graphLookup/>` stage has a strict
+   memory limit of 100 megabytes and ignores the ``allow_disk_use`` field.
+
+.. _cpp-aggregation-example:
+
+Aggregation Example
+-------------------
+
+.. note::
+  
+   The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+   database from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+   free MongoDB Atlas cluster and load the sample datasets, see the :atlas:`Get Started with Atlas
+   </getting-started>` guide.
+
+To perform an aggregation, pass a ``mongocxx::pipeline`` instance containing the aggregation
+stages to the ``collection.aggregate()`` method.
+
+The following code example produces a count of the number of bakeries in each borough
+of New York. To do so, it uses an aggregation pipeline that contains the following stages:
+
+- :manual:`$match </reference/operator/aggregation/match/>` stage to filter for documents
+  in which the ``cuisine`` field contains the value ``"Bakery"``
+
+- :manual:`$group </reference/operator/aggregation/group/>` stage to group the matching
+  documents by the ``borough`` field, accumulating a count of documents for each distinct
+  value
+
+.. io-code-block::
+
+   .. input:: /includes/aggregation.cpp
+      :start-after: start-match-group
+      :end-before: end-match-group
+      :language: cpp
+      :dedent:
+
+   .. output:: 
+
+      { "_id" : "Brooklyn", "count" : 173 }
+      { "_id" : "Queens", "count" : 204 }
+      { "_id" : "Bronx", "count" : 71 }
+      { "_id" : "Staten Island", "count" : 20 }
+      { "_id" : "Missing", "count" : 2 }
+      { "_id" : "Manhattan", "count" : 221 }
+
+Explain an Aggregation
+~~~~~~~~~~~~~~~~~~~~~~
+
+To view information about how MongoDB executes your operation, you can
+instruct MongoDB to **explain** it. When MongoDB explains an operation, it returns
+**execution plans** and performance statistics. An execution
+plan is a potential way MongoDB can complete an operation.
+When you instruct MongoDB to explain an operation, it returns both the
+plan MongoDB executed and any rejected execution plans.
+
+To explain an aggregation operation, run the ``explain`` database command by specifying
+the command in a BSON document and passing it as an argument to the ``run_command()``
+method.
+
+The following example performs the same aggregation operation as the preceding
```

Comment:
S: When I read this, I expected the return result to be both the operation explanation as well as the previous result of how many bakeries were in each borough. 

You could potentially clarify by saying something like: "The following example performs the same aggregation operation as the preceding example, but instead of producing a count of the bakeries in each borough of New York, the database command explains the operation."

Reply:
Reworded to something a little different - lmk what you think